packages/starlight/components/AnchorHeading.astro

```diff
@@ -0,0 +1,55 @@
+---
+import type { HTMLAttributes } from 'astro/types';
+import { transform } from 'ultrahtml';
+import sanitize from 'ultrahtml/transformers/sanitize';
+// These styles are included globally by default, but can be removed when `markdown.headingLinks` is
+// set to `false`. We import them here to ensure they are included if the component is used with the
+// global Markdown option disabled.
+import '../style/anchor-links.css';
+import { AstroError } from 'astro/errors';
+
+const headingLevels = [1, 2, 3, 4, 5, 6, '1', '2', '3', '4', '5', '6'] as const;
+type HeadingLevel = (typeof headingLevels)[number];
+interface Props extends HTMLAttributes<'h1'> {
+	level: HeadingLevel;
+	id: string;
+}
+
+const { level, id, ...attrs } = Astro.props;
+
+if (!id) {
+	throw new AstroError(
+		'Missing `id` attribute passed to `<AnchorHeading>` component',
+		`The \`<AnchorHeading>\` component requires an \`id\` attribute, but received \`${typeof id === 'string' ? '""' : id}\`.`
```

Comment:
The `${typeof id === 'string' ? '""' : id}` part seems confusing? Is the idea to have it show what it received if it's not a string? If that's the case, with the `if (!id)` condition, I guess it would only hit that scenario if the value is not a string and is something like `0`?

Reply:
The intention here is to display a good empty string representation:
| Input | Output |
|--------|--------|
| `""` | `""` |
| `0` | `0` |
| `null` | `null` |
| `undefined` | `undefined` | 
| `false` | `false` | 
For most false-y values `toString()` returns a good representation but empty string is just empty string.

Reply:
Ah I see, went on the wrong track of thought thinking you wanted to show all invalid values 👍